src/gax-internal/tests/http_retry_loop.rs

```diff
@@ -82,13 +85,24 @@ mod tests {
         Ok(())
     }
 
+    #[test_case(false; "tracing off")]
+    #[test_case(true; "tracing on")]
     #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
-    async fn retry_loop_retry_success() -> Result<()> {
+    async fn retry_loop_retry_success(enable_tracing: bool) -> Result<()> {
+        if enable_tracing && !cfg!(feature = "_unstable-o12y") {
```

Comment:
Ditto, this is too much logic for a test. In addition, this suggests the test should be conditionally compiled.